spring/src/main/java/org/axonframework/spring/eventsourcing/SpringPrototypeAggregateFactory.java

```diff
@@ -88,15 +111,36 @@ public void afterPropertiesSet() {
                                    + "The bean with name '%s' does not have the 'prototype' scope.",
                            beanName, prototypeBeanName));
         }
-        this.delegate = new AbstractAggregateFactory<T>(getAggregateType()) {
+        AggregateModel<T> model;
+        if (applicationContext.getBeanNamesForType(Configuration.class).length > 0) {
+            Configuration configuration = applicationContext.getBean(Configuration.class);
+            model = AnnotatedAggregateMetaModelFactory.inspectAggregate(getAggregateType(),
+                                                                        configuration.parameterResolverFactory(),
+                                                                        configuration
+                                                                                .handlerDefinition(getAggregateType()),
+                                                                        subtypes.keySet());
+        } else {
+            model = AnnotatedAggregateMetaModelFactory.inspectAggregate(getAggregateType(),
+                                                                        subtypes.keySet());
+        }
+        this.delegate = new AbstractAggregateFactory<T>(model) {
             @Override
             protected T doCreateAggregate(String aggregateIdentifier, DomainEventMessage firstEvent) {
-                return (T) applicationContext.getBean(prototypeBeanName);
+                return (T) applicationContext.getBean(prototype(firstEvent.getType()));
+            }
+
+            private String prototype(String firstEventType) {
```

Comment:
Confusing parameter name. The type is that of the aggregate, not the event. The fact that it's the type derived from the first event, it not relevant in this scope anymore. I suggest using "aggregateType" instead.
Also, the method should allow inexistent types if no substypes are declared in the aggregate, to ensure compatibility with 4.2, where aggregate type wasn't used to initialize an aggregate.